src/main/java/br/unb/cic/analysis/ioa/InterproceduralOverrideAssignment.java

```diff
@@ -1,56 +1,299 @@
 import soot.*;
+import soot.jimple.AssignStmt;
+import soot.jimple.InvokeStmt;
+import soot.jimple.internal.JInstanceFieldRef;
+import soot.toolkits.scalar.ArraySparseSet;
+import soot.toolkits.scalar.FlowSet;
 
 import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
 
-public class InterproceduralOverrideAssignment extends SceneTransformer  {
-
-    private Set<SootMethod> visitedMethods;
+// TODO Add treatment of if, loops ... (ForwardFlowAnalysis)
+// TODO Do not add anything when assignments are equal.
+public class InterproceduralOverrideAssignment extends SceneTransformer implements AbstractAnalysis {
 
+    private Set<Conflict> conflicts;
     private PointsToAnalysis pta;
-
     private AbstractMergeConflictDefinition definition;
+    private FlowSet<DataFlowAbstraction> res;
+    private Body body;
+
+    private Logger logger;
 
     public InterproceduralOverrideAssignment(AbstractMergeConflictDefinition definition) {
-        visitedMethods = new HashSet<>();
+        this.conflicts = new HashSet<>();
         this.definition = definition;
+        this.res = new ArraySparseSet<>();
+
+        this.logger = Logger.getLogger(
+                InterproceduralOverrideAssignment.class.getName());
+    }
+
+    @Override
+    public void clear() {
+        conflicts.clear();
+    }
+
+    @Override
+    public Set<Conflict> getConflicts() {
+        return conflicts;
+    }
+
+    private void configureEntryPoints() {
+        List<SootMethod> entryPoints = new ArrayList<>();
+        definition.getSourceStatements().forEach(s -> entryPoints.add(s.getSootMethod()));
+        Scene.v().setEntryPoints(entryPoints);
     }
 
     @Override
     protected void internalTransform(String s, Map<String, String> map) {
         definition.loadSourceStatements();
         definition.loadSinkStatements();
+        List<SootMethod> traversedMethods = new ArrayList<>();
 
         configureEntryPoints();
 
         List<SootMethod> methods = Scene.v().getEntryPoints();
-        pta = Scene.v().getPointsToAnalysis();
-        methods.forEach(m -> traverse(m));
+        this.pta = Scene.v().getPointsToAnalysis();
+        methods.forEach(m -> traverse(m, traversedMethods, Statement.Type.IN_BETWEEN));
+
+        String stringConflicts = String.format("%s", conflicts);
+        logger.log(Level.INFO, stringConflicts);
     }
 
-    private void traverse(SootMethod m) {
-        if(visitedMethods.contains(m) || m.isPhantom()) {
+    /**
+     * This method captures the safe body of the current method and delegates the analysis function to units of (LEFT or RIGHT) or units of BASE.
+     *
+     * @param sm            name of the current method to be traversed;
+     * @param traversed     list of methods that have already been traversed;
+     * @param flowChangeTag This parameter identifies whether statement is in the flow of any statements already marked.
```

Comment:
@barbosamaatheus "whether statement" faria sentido escrever isso se statement fosse um parametro, mas nao eh. "whether statements in sm" seria isso? mesmo assim tá confusa a explicação

Reply:
O objeto que tem as marcações Source e Sink é o Statement (Nossa implementação). No texto eu me referencio a ele. Esse Statement é criado no método `getStatementAssociatedWithUnit` O que está confundindo ai é a nomenclatura dos objetos.

Reply:
eu entendo. a questão é que a explicação nao esta clara e nao vai ser entendida por outras pessoas